### ISSUE TITLE
Difference in the way messages look on different devices and OS 

### ISSUE THREAD

**Issue by agnieszkakowalska** [EDMdesigner/email-html-generator-issues#21]
Our client noticed, that messages look differently on some devices. For example:

EDM Project ID: 57ff9f9e3031fd52085202ec
api: API_EXPERTSENDER
user: Unit_1504

I tried to open this message on different OS and devices and this is how it looks like:

CORRECTLY (like on desktop browser):

OS: IOS 7.1.2;
Email provider: Gmail (browser);
Browser version: Safari
Screnshot name: Browser_Gmail_IOS_1; Browser_Gmail_IOS_2; Browser_Gmail_IOS_3

OS: Android 5.0.1;
Email provider: Yandex (application);
Application version: 6.10.9.136347105.release;
Screnshot name: Application_Yandex_Android_1

OS: Android 5.0.1;
Email provider: Yandex (browser);
Browser version: Chrome 54.0.2840.68
Screnshot name: Browser_Yandex_Android_1

INCORRECTLY:

OS: IOS 7.1.2;
Email provider: Gmail (application);
Application version: 4.4.0
Screnshot name: Application_gmail_IOS_1; Application_gmail_IOS_2; Application_gmail_IOS_3

OS: Android 5.0.1;
Email provider: Gmail (application);
Application version: 6.10.9.136347105.release;
Screnshot name: Application_Gmail_Android_1; Application_Gmail_Android_2

OS: Android 5.0.1;
Email provider: Gmail (browser);
Browser version: Chrome 54.0.2840.68
Screnshot name: Browser_Gmail_Android_1; Browser_Gmail_Android_2; Browser_Gmail_Android_3

Can you suggest why is it so? Is it a bug or is this something that cannot be fixed? What can we do to make messages look the same way on every device? Customers need this from us.

Screenshots are in attachment.
[screenshots_EDM.zip](https://github.com/EDMdesigner/email-html-generator-issues/files/599371/screenshots_EDM.zip)

**Comment by rolficak** [comment]
Hi Agnieszka,

We don't support Yandex.ru - http://edmdesigner.com/compatibility
Browser view on Android or iOS is neither on the list.
Based on the screenshot they don't support responsive CSS.

Gmail recently started supporting responsive design, but on iOS updates are
late, that might cause the issues you show on iOS.

So there's nothing we could fix on these screenshots right now.
ᐧ

Roland Pokornyik
CEO, Co-founder @ EDMdesigner.com

+36-70-608-4433

_Schedule http://www.meetme.so/Roland-EDMdesigner a Skype chat with me!_

On Fri, Nov 18, 2016 at 9:35 AM, agnieszkakowalska <notifications@github.com

**Comment by agnieszkakowalska** [comment]
Hi Roland, Thanks for the reply. Is it possible to add support for Mail.ru app and Yandex in the future?

**Comment by rolficak** [comment]
Do you know of any online test solutions (like Litmus) which would provide
inbox preview of these email clients?

ᐧ

Roland Pokornyik
CEO, Co-founder @ EDMdesigner.com

+36-70-608-4433

*Schedule <http://www.meetme.so/Roland-EDMdesigner> a Skype chat with me!*

On Mon, Dec 12, 2016 at 2:49 PM, agnieszkakowalska <notifications@github.com

**Comment by agnieszkakowalska** [comment]
Isn't Mail.ru available in Litmus? https://litmus.com/blog/mail-ru-previews-now-available-in-litmus

**Comment by gyulanemeth** [comment]
Yes, Mail.ru is available in Litmus. Unfortunately Yandex is not.

**Comment by agnieszkakowalska** [comment]
So will you add support for Mail.ru? When can you add it? I don't know if Yandex is available in any other client.

**Comment by agnieszkakowalska** [comment]
Hi, can you answer my questions above?

**Comment by rolficak** [comment]
It's not on our short term roadmap, but we'll look into it.

Mail.ru is one of the trickiest email clients as far as I know.
ᐧ

Roland Pokornyik
CEO, Co-founder @ EDMdesigner.com

+36-70-608-4433

*Schedule <http://www.meetme.so/Roland-EDMdesigner> a Skype chat with me!*

On Fri, Jan 13, 2017 at 2:54 PM, agnieszkakowalska <notifications@github.com

**Comment by agnieszkakowalska** [comment]
Mail.ru and Yandex support is really important for us as we have a huge Russian market - can you say more or less when it could be added? Or any minimum time required for working on it? Is adding Yandex also doable?

**Comment by ughoo** [comment]
Sorry for late reply. I need to check those clients in details, and to find the proper way to support, - if it's possible. I'll reply on this topic, with the results, as soon as possible.

**Comment by ughoo** [comment]
I did the tests on Mail.ru and Yandex.ru, both of them are stable.
(They leave CSS in the html, so border-radius, and other stylish things also works.)

Please let me know, if you have any bugs on these webclients!

**Comment by agnieszkakowalska** [comment]
Hi @ughoo, we made several tests with the settings of different elements of the message and the worst problems were solved. Now the main problem is the font size. In some applications it is too small for comfortable reading (like on the screenshot called SMALL_FONT.jpg). In other applications the size of the message is too big and there is no ability to see the whole message on the screen of a phone (BIG_MESSAGE.jpg). But also there applications in which messages look well because text adapts to the screen size of the phone (OK.jpg). In the image below you can find how messages look in different apps:

![obraz](https://cloud.githubusercontent.com/assets/17157918/22370058/b3ac0ba6-e48f-11e6-93fe-d85f907f2a64.png)

So, here are the most imortant things for our customers:
1) to make messages look like on the screenshot named OK.jpg (font size must be adaptive to the screen size) in applications of Yandex and Mailru for the last versions of Android and iOS;
2) to make messages look like on the screenshot named OK.jpg (font size must be adaptive to the screen size) in different browsers for Yandex.ru and Mail.ru for the last versions of Android and iOS;
3) to make messages look like on the screenshot named OK.jpg (font size must be adaptive to the screen size) in different browsers for Gmail.com for the last versions of Android and iOS.

SMALL_FONT:
![small_font](https://cloud.githubusercontent.com/assets/17157918/22370096/e09f70a8-e48f-11e6-9287-4eaf8d5a4ad8.jpg)

BIG_MESSAGE:
![big_message](https://cloud.githubusercontent.com/assets/17157918/22370098/e0a14bc6-e48f-11e6-9020-8ec9406626db.jpg)

OK:
![ok](https://cloud.githubusercontent.com/assets/17157918/22370097/e0a10bf2-e48f-11e6-89de-af4e09be7d39.jpg)

**Comment by ughoo** [comment]
So it means, that Yandex and Mailru apps and mobile versions of webmail in mobile browsers shows desktop version...with two different behavior. This only can happen, when those apps and webmails are ignores all media queries included in our html. - Without Media Queries, we are not able to suppport proper view in those email clients.

If there is any documented clue, how these apps can handle media queries, or custom targeted css, then we are able to research and implement a method for this problem.
